### PR TITLE
#147 Change calendar setting pages to dialog boxes

### PR DESCRIPTION
**Pull request by pbson**
Fixed: https://github.com/OpenPaaS-Suite/esn-frontend-calendar/issues/147
Demo: https://streamable.com/nhaphp

### CODE REVIEW

File: src/linagora.esn.calendar/app/settings/settings.controller.js [hunk 1]
```diff
@@ -3,6 +3,22 @@
...
+  var self = this;
+  self.reloadCalendarMainState = reloadCalendarMainState;
+
+  ////////
+
   $scope.hasTouchscreen = touchscreenDetectorService.hasTouchscreen();
-}
+
+  function reloadCalendarMainState(){
+    console.log($state);
```
Comment (rezk2ll) [inline src/linagora.esn.calendar/app/settings/settings.controller.js:15]:
```suggestion
```

File: src/linagora.esn.calendar/app/settings/settings.controller.js [hunk 2]
```diff
@@ -3,6 +3,22 @@
...
+    if ($state.current.name === 'calendar.main.settings.display'){
+      $state.go('calendar.main', {}, {
+        reload: 'calendar.main',
+      });
+    }else{
+      $state.go('calendar.main');
+    }
```
Comment (rezk2ll) [inline src/linagora.esn.calendar/app/settings/settings.controller.js:22]:
```suggestion
    $state.current.name === 'calendar.main.settings.display' ?
      $state.go(CAL_MODULE_METADATA.homePage, {}, { reload: CAL_MODULE_METADATA.homePage }):
      $state.go(CAL_MODULE_METADATA.homePage);
```

File: src/linagora.esn.calendar/app/settings/settings.controller.js [hunk 3]
```diff
@@ -3,6 +3,22 @@
...
+  }
+}
```
Comment (rezk2ll) [inline src/linagora.esn.calendar/app/settings/settings.controller.js:24]:
```suggestion
}

```

File: src/linagora.esn.calendar/app/settings/settings.less [hunk 4]
```diff
@@ -7,4 +7,4 @@
     color: @secondaryTextColor;
     font-size: 18px;
   }
-}
+}
```
Comment (rezk2ll) [inline src/linagora.esn.calendar/app/settings/settings.less:10]:
```suggestion
}

```

File: src/linagora.esn.calendar/app/settings/sidebar-button/settings-sidebar-button.pug [hunk 5]
```diff
@@ -1,4 +1,4 @@
 .cal-settings-sidebar-button
-  .lv-item.clickable(href="", ui-sref="calendar.settings")
+  .lv-item.clickable(href="", ui-sref="calendar.main.settings")
     i.mdi.mdi-settings
     | {{ 'Manage calendars' | translate }}
```
Comment (rezk2ll) [inline src/linagora.esn.calendar/app/settings/sidebar-button/settings-sidebar-button.pug:4]:
```suggestion
    | {{ 'Manage calendars' | translate }}

```

File: src/linagora.esn.calendar/app/settings/settings.pug [hunk 6]
```diff
@@ -1,8 +1,19 @@
...
+          li.btn-wave(ui-sref-active="active")
+            a(href="", ui-sref="calendar.main.settings.calendars", ui-sref-opts="{ location: 'replace' }") {{ "Calendars" | translate }}
+          li.btn-wave(ui-sref-active="active")
+            a(href="", ui-sref="calendar.main.settings.display", ui-sref-opts="{ location: 'replace' }") {{ "Display" | translate }}
+        ui-view.configuration-content(name="settings", noanimation="noanimation")
+      .modal-footer.flex
+        .flex-vertical-centered.flex-end.hide-sm.hide-xs
+          button.btn.btn-link.color-default.close-button(type='button', ng-click="$ctrl.reloadCalendarMainState()") {{ 'Close' | translate }}
+        .flex-vertical-centered.flex-start.hide-gt-sm
+          button.btn.btn-link.color-default.close-button(type='button', ng-click="$ctrl.reloadCalendarMainState()") {{ 'Close' | translate }}
```
Comment (rezk2ll) [inline src/linagora.esn.calendar/app/settings/settings.pug:19]:
```suggestion
          button.btn.btn-link.color-default.close-button(type='button', ng-click="$ctrl.reloadCalendarMainState()") {{ 'Close' | translate }}

```

File: src/linagora.esn.calendar/app/settings/settings.pug [hunk 7]
```diff
@@ -1,8 +1,19 @@
...
-    li.btn-wave(ui-sref-active="active")
-      a(href="", ui-sref="calendar.settings.calendars", ui-sref-opts="{ location: 'replace' }") {{ "Calendars" | translate }}
-    li.btn-wave(ui-sref-active="active")
-      a(href="", ui-sref="calendar.settings.display", ui-sref-opts="{ location: 'replace' }") {{ "Display" | translate }}
-  ui-view.configuration-content(name="settings", noanimation="noanimation")
+.modal.center.full-screen(tabindex="-1", role="dialog")
+  .modal-dialog
+    .modal-content
+      .modal-header
+        button.close(type='button', ng-click="$ctrl.reloadCalendarMainState()") &times;
```
Comment (rezk2ll) [inline src/linagora.esn.calendar/app/settings/settings.pug:6]:
```suggestion
        button.close(type='button', ng-click="$ctrl.closeSettingsModal()") &times;
```

the function name can be renamed to express what the button does.

File: src/linagora.esn.calendar/app/settings/calendars/settings-calendars.less [hunk 8]
```diff
@@ -1 +1,7 @@
 @import "./item/settings-calendars-item.less";
+
+.calendar-list {
+    .no-shared-calendar{
```
Comment (huy-ta) [inline src/linagora.esn.calendar/app/settings/calendars/settings-calendars.less:4]:
```suggestion
    .no-shared-calendar {
```

File: src/linagora.esn.calendar/app/settings/display/settings-display.pug [hunk 9]
```diff
@@ -1,15 +1,12 @@
 // TODO: Revert to use the mixin here in case we can fix the translation issue. Refer to https://github.com/OpenPaaS-Suite/esn-frontend-calendar/issues/42#issuecomment-672757911 for some suggestions.
-sub-header
-  cal-settings-display-subheader(submit="ctrl.submit()", form="form")
-
 form(name="form")
   h3 {{ "Non-working days" | translate }}
   label.checkbox.checkbox-inline.flex
-    input(type="checkbox", ng-model="ctrl.configurations.workingDays")
+    input(type="checkbox", ng-model="ctrl.configurations.workingDays",ng-change="ctrl.submit()")
```
Comment (huy-ta) [inline src/linagora.esn.calendar/app/settings/display/settings-display.pug:5]:
```suggestion
    input(type="checkbox", ng-model="ctrl.configurations.workingDays", ng-change="ctrl.submit()")
```

File: src/linagora.esn.calendar/app/calendar-configuration/calendar-configuration.less [hunk 10]
```diff
@@ -22,3 +22,21 @@
     margin-top: 25px;
   }
 }
+
+.delete-button{
```
Comment (huy-ta) [inline src/linagora.esn.calendar/app/calendar-configuration/calendar-configuration.less:26]:
```suggestion
.delete-button{
```

:trollface:

**Review by huy-ta — APPROVED** [review]

**Review by rezk2ll — APPROVED** [review]

**Pull request merged**